assets/dev/js/editor/document/history/commands/base/base.js

```diff
@@ -9,4 +9,29 @@ export default class Base extends CommandInternalBase {
 		 */
 		this.history = elementor.documents.getCurrent().history;
 	}
+
+	onBeforeApply( args = {} ) {
+		super.onBeforeApply( args );
+
+		if ( ! this.useHistory() ) {
+			throw new $e.modules.HookBreak;
+		}
+	}
+
+	onCatchApply( e ) {
+		if ( ! ( e instanceof $e.modules.HookBreak ) ) {
+			super.onCatchApply( e );
+		}
```

Comment:
Why?

Reply:
The common assumption of using both `start/end-log` commands and `CommandHistory` is that they're using this base class. If we check for the `useHistory` of the first currently running command here, we can determine whether to use history even in nested commands, and also whenever using `start/end-log` actions.

Reply:
I think that everyone that manually handle the _History_  should somehow know what the status of history, the infa that should remain the same... the only code that should stay in `CommandHistory` is the one that check first args trace in the `consturctor`.

Reply:
Moved to `CommandHistory`.

Reply:
?